### ISSUE TITLE
E206 Variable Space: Fix nested JSON object false positive

### ISSUE THREAD

**Issue by simonkeyd** [ansible/ansible-lint#665]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible 2.8.5
ansible-lint 4.2.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

When using compact form nested JSON object within a Jinja2 context as shown in the following example:
```yaml
set_fact:"{{ {'test': {'subtest': variable}} }}"
```
'__variable}}__' should not raise a 206 error since we are only closing a JSON object.

# Actual Behaviour

`206-false-positive.yml`:
```yaml
---
- hosts: localhost
  connection: local
  gather_facts: false
  vars:
    variable_example: value
  tasks:
    - name: Set 'test_obj' json object
      set_fact:
        test_obj: {}

    - name: Append a nested object to 'test_obj'
      set_fact:
        test_obj: "{{ {'test_nested_obj': {'test': variable_example}} | combine(test_obj) }}"
```
```sh
$ ansible-lint 206-false-positive.yml
[206] Variables should have spaces before and after: {{ var_name }}
206-false-positive.yml:14
        test_obj: "{{ {'test_nested_obj': {'test': variable_example}} | combine(test_obj) }}"
```

**Comment by simonkeyd** [comment]
#664
